Gems/Multiplayer/Code/Source/Components/NetworkTransformComponent.cpp

```diff
@@ -93,9 +103,19 @@ namespace Multiplayer
         targetTransform.SetUniformScale(GetScale());
 
         // Hard set the entities transform
-        if (!GetTransformComponent()->GetWorldTM().IsClose(targetTransform))
+        if (GetParentEntityId() == InvalidNetEntityId)
         {
-            GetTransformComponent()->SetWorldTM(targetTransform);
+            if (!GetTransformComponent()->GetWorldTM().IsClose(targetTransform))
+            {
+                GetTransformComponent()->SetWorldTM(targetTransform);
+            }
+        }
+        else
+        {
+            if (!GetTransformComponent()->GetLocalTM().IsClose(targetTransform))
+            {
+                GetTransformComponent()->SetLocalTM(targetTransform);
+            }
```

Comment:
I doubt it leads to any improvement, since it's a simple getter. It should be inlining without any issues:

```
    AzFramework::TransformComponent* NetworkTransformComponentBase::GetTransformComponent() 
    {
        return m_transformComponent;
    }
```

Reply:
The inline will only happen in lucky unity builds, since that function is defined in auto-generated `Generated\Source\AutoGen\NetworkTransformComponent.AutoComponent.cpp` ?

Reply:
Huh, you are right. I checked and in my profile build it doesn't get inlined. :( But it should inline for sure in release builds. I'll apply your suggestion, though.